pkg/apis/messaging/v1alpha1/nats_jetstream_types.go

```diff
@@ -0,0 +1,98 @@
+	eventingduckv1 "knative.dev/eventing/pkg/apis/duck/v1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+)
+
+// +genclient
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NatsJetStreamChannel is a resource representing a NATS JetStream Channel.
+type NatsJetStreamChannel struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the Channel.
+	Spec NatsJetStreamChannelSpec `json:"spec,omitempty"`
+
+	// Status represents the current state of the NatssChannel. This data may be out of
+	// date.
+	// +optional
+	Status NatsJetStreamChannelStatus `json:"status,omitempty"`
+}
+
+// Check that Channel can be validated, can be defaulted, and has immutable fields.
+var (
+	_ apis.Validatable = (*NatsJetStreamChannelSpec)(nil)
+	_ apis.Defaultable = (*NatsJetStreamChannelSpec)(nil)
+	// Check that InMemoryChannel can return its spec untyped.
+	_ apis.HasSpec = (*NatsJetStreamChannel)(nil)
+	// Check that we can create OwnerReferences to an InMemoryChannel.
+	_ kmeta.OwnerRefable = (*NatsJetStreamChannel)(nil)
+	_ runtime.Object     = (*NatsJetStreamChannel)(nil)
+	_ duckv1.KRShaped    = (*NatsJetStreamChannel)(nil)
+)
+
+// NatsJetStreamChannelSpec defines the specification for a NatssChannel.
+type NatsJetStreamChannelSpec struct {
+	// inherits duck/v1 ChannelableSpec, which currently provides:
+	// * SubscribableSpec - List of subscribers
+	// * DeliverySpec - contains options controlling the event delivery
+	eventingduckv1.ChannelableSpec `json:",inline"`
+}
+
+// NatsJetStreamChannelStatus represents the current state of a NatssChannel.
+type NatsJetStreamChannelStatus struct {
+	// inherits duck/v1 ChannelableStatus, which currently provides:
+	// * ObservedGeneration - the 'Generation' of the Service that was last processed by the controller.
+	// * Conditions - the latest available observations of a resource's current state.
+	// * AddressStatus is the part where the Channelable fulfills the Addressable contract.
+	// * Subscribers is populated with the statuses of each of the Channelable's subscribers.
+	// * DeadLetterChannel is a KReference and is set by the channel when it supports native error handling via a channel
+	//   Failed messages are delivered here.
+	eventingduckv1.ChannelableStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NatsJetStreamChannelList is a collection of NatssChannels.
+type NatsJetStreamChannelList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NatsJetStreamChannel `json:"items"`
+}
+
+// GetGroupVersionKind returns GroupVersionKind for NatssChannels
+func (*NatsJetStreamChannel) GetGroupVersionKind() schema.GroupVersionKind {
+	return SchemeGroupVersion.WithKind("NatssChannel")
```

Comment:
Shouldn't this be the Jetstream channel and not NatssChannel?

Reply:
yeah, `NatsJetStreamChannel`